sysml.library/Domain Libraries/Geometry/ShapeItems.sysml

```diff
@@ -0,0 +1,249 @@
+package ShapeItems {
+	private import ScalarValues::String;
+	private import ScalarValues::Boolean;
+	private import UnitsAndScales::CoordinateTransformation;
+	private import ISQ::*;
+	private import SI::*;
+	private import Collections::KeyValuePair;
+	private import Collections::Map;
+	private import Objects::*;
+	private import Items::*;
+	private import SequenceFunctions::union;
+	private import CollectionFunctions::contains;
+	private import SequenceFunctions::isEmpty;
+	connection def WithinBoth :> Occurrences::WithinBoth { }
+	
+	/**
+	 * Most general shape.	Solids are objects with these as space boundary.	
+	 */
+	abstract item def Shell :> Item, StructuredSpaceObject, Surface	 {
+	}
+
+	/**
+	* Straight line.
+	*/
+	item def Line :> Curve {
+		attribute :>> length;
+
+		attribute :>> outerSpaceDimension = 1;
+	}
+
+	/**
+	 * Flat surface.
+	 */
+	item def PlanarSurface :> Surface {
+		attribute :>> area;
+		attribute :>> outerSpaceDimension = 2;
+	}
+
+	item def Circle :> Shell {
+		attribute radius :>> radius;
+
+		item :>> faces [0];
+		item :>> edges [1] {
+			attribute circumference :>> length = Circle::radius * 3.1415926535898;
+			item :>> spaceBoundary : Item [0];
+			item :>> vertices [0];
+		}
+		item :>> vertices [0];
+	}
+
+	/**
+	 * Surface bounded by a circle.
+	 */
+	item def Disc :> Shell {
+		attribute radius :>> radius;
+
+		item :>> spaceBoundary : Circle [1]
```

Comment:
If a Circle and a Disc are both Shells (Surfaces), then what is the difference? How can a Disc, as a Surface, be bounded by a Circle, if that is also a Surface? Shouldn't a Circle be a Curve without a boundary, not a Surface?

Reply:
Shells can be innerDim 1 by not having faces, as in Circle (Shell doesn't restrict StructuredSpaceObject, just renames it).  Perhaps we have a naming or factoring problem.  @hpdekoning, is there a common name for 1-dimensional "shells", like polygons, ie, curves that are potentially segmented into edges and vertices?

Reply:
> Oops, just noticed Shells are Surfaces, that's not right, or at least not consistent.

Yes, that was my point. 

Colloquially, at last, it makes sense to me to have a Shell be a surface. But then it doesn't make sense to have a Circle be a Shell. Per your previous comment, it would be nice to have a separate name for one-dimensional "shells". In the shape model, are these curves always going to be closed (like circles and polygons)? Could you repurpose "Line" for that, and rename the current "Line" to "StraightLine"? Or maybe "Path"?

Reply:
See proposal at HP and I just posted at [ST6RI-507](https://openmbee.atlassian.net/browse/ST6RI-507?focusedCommentId=13373) would be Shell of 2 dim and Path for 1 dim.  These could be open or closed.  Would that be good?

Reply:
Fine with me.

Reply:
Done.